pkgs/development/python-modules/pleroma-bot/default.nix

```
@@ -0,0 +1,36 @@
+{ lib
+, buildPythonPackage
+, fetchFromGitHub
+, pytestCheckHook
+, requests-mock
+, oauthlib
+, requests_oauthlib
+, requests
+, pyaml
+}:
+
+buildPythonPackage rec {
+  pname = "pleroma-bot";
+  version = "0.8.6";
+
+  src = fetchFromGitHub {
+    owner  = "robertoszek";
+    repo   = "pleroma-bot";
+    rev    = "0.8.6";
+    sha256 = "1q0xhgqq41zbqiawpd4kbdx41zhwxxp5ipn1c2rc8d7pjyb5p75w";
+  };
+
+  checkInputs = [ pytestCheckHook ];
+
+  buildInputs = [ requests-mock ];
```

`requests-mock` is a [test requirement](https://github.com/robertoszek/pleroma-bot/blob/master/setup.py#L51).

```suggestion
  checkInputs = [ pytestCheckHook requests-mock ];
```

---

Thanks for the explanation, your guidance has been immensely helpful 👍